corehq/apps/users/templates/users/commcare_user_confirm_account.html

```diff
@@ -1,39 +1,106 @@
-  </div>
-{% endblock %}
+{% block title %}{% blocktrans %}Confirm your account to join the {{ domain_name }} project{% endblocktrans %}{% endblock title %}
+{% block tabs %}{% endblock %}
 
-{% block content %}
-  {% if user.is_account_confirmed and user.is_active %}
-    <div>
-      <p>
-        {% blocktrans %}
-          Your account is already confirmed! Go sign in.
-        {% endblocktrans %}
-      </p>
-    </div>
-  {% elif user.is_account_confirmed %}
-    <div class="alert-alert-danger">
-      <p>
-        {% blocktrans %}
-          It looks like your account has been deactivated.
-          Contact your administrator if you think this is a mistake.
-        {% endblocktrans %}
-      </p>
-    </div>
-  {% else %}
-    <div id="confirm-account-form">
-      <p>Confirm your account here!</p>
-      {#  todo    {% crispy form %}#}
+{% block login-content %}
+  <div class="container-fluid">
+    <div class="row">
+      <div class="col-xs-12">
+        <div class="reg-form-container sign-in-container">
+          <div class="form-bubble form-bubble-lg">
+            {% if request.user.is_authenticated %}
+              <h3>
+                {% trans "You are already signed in." %}
+              </h3>
+              <h4>
+                {% url 'logout' as sign_out_url %}
+                {% blocktrans %}
+                  To accept this invitation, you must first <a href="{{ sign_out_url }}">sign out</a>.
+                {% endblocktrans %}
+              </h4>
+            {% elif user.is_account_confirmed and user.is_active %}
+              <h3>
+                {% trans "Your account is already confirmed!" %}
+              </h3>
+              <h4>
+                {% url 'login' as login_url %}
+                {% blocktrans %}
+                  <a href="{{ login_url }}">Login</a> to access your account.
+                {% endblocktrans %}
+              </h4>
+            {% elif user.is_account_confirmed %}
+              <h3>
+                {% blocktrans %}
+                  It looks like your account has been deactivated.
+                  Contact your administrator if you think this is a mistake.
+                {% endblocktrans %}
+              </h3>
+            {% else %}
+              <h2>
+                {% blocktrans %}Confirm Your Account{% endblocktrans %}
+              </h2>
+              <div class="help-block">
+                {% blocktrans %}
+                  Fill in the details below to confirm your account.
+                {% endblocktrans %}
+              </div>
+              <form name="form" method="post" action="">
+                <h4>
+                  {% with user.raw_username as username %}
+                  {% blocktrans %}
+                    Your username <strong>{{ username }}</strong> was chosen by your project administrator.
+                  {% endblocktrans %}
+                  {% endwith %}
+                </h4>
+                {% csrf_token %}
```

Comment:
a fair amount of this part is duplicated [from here](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/users/templates/users/accept_invite.html#L97-L123) (which is also mostly duplicated [here](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/domain/templates/login_and_password/password_reset_confirm.html#L16-L38) and [here](https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/domain/templates/login_and_password/partials/password_reset_form_only.html#L4-L30)).

Again, unclear whether it's worth figuring out a way to share them despite being slightly different.

Reply:
My fuzzy metric is that 10-20 lines is about where HTML is worth de-duplicating. I haven't combed through every line, but the part you linked looks identical except for the `{% if create_domain %}<legend>...</legend>{% endif %}`, which could probably be removed altogether from `accept_invite.html`.

Reply:
https://github.com/dimagi/commcare-hq/pull/26917